allow module to work with puppet 6.16 and higher

applewiskey opened the pull request:
remove url encoding on new puppet agent versions as it's done internally

applewiskey committed:
allow module to work with puppet 6.16 and higher

remove url encoding on new puppet agent versions as it's done internally

applewiskey committed:
fix puppet 6 compatibility

it resolved that face and function have different internal processing
face does noe require URL encoding and funtions do

raphink commented:
FTR, `puppetdb_query()` is now a standard function that comes with the PuppetDB termini package: https://puppet.com/docs/puppetdb/5.2/api/query/tutorial.html

applewiskey commented:
Hi, @raphink .
Well, it is, but if you have tens of thousands line of code with 10-100 usages for query_nodes written by other teams you have to choose between fixing code for everyone, ask them to fix and wait a year or fixing the source. So I decided to do the last. Even if @dalen will never merge it, someone with the same issues can find it and use.

raphink commented:
@applewiskey another option for you would be to use `puppet-lint` to automate the migration. Writing a `puppet-lint` plugin allows you to detect usage of the `query_nodes` or `query_facts` and you can even write a `fix` method to automatically convert to `puppetdb_query()`. See for example https://github.com/voxpupuli/puppet-lint-absolute_classname-check/blob/master/lib/puppet-lint/plugins/check_absolute_classname.rb#L47

sharewax committed:
Fix warning about using 'Puppet::Network::HttpPool.http_instance'

    Fix deprecation warning:
    Puppet The method 'Puppet::Network::HttpPool.http_instance' is deprecated. Use Puppet.runtime[:http] instead

sharewax committed:
Merge pull request #1 from wgsateam/sharewax-patch-1

Fix warning about using 'Puppet::Network::HttpPool.http_instance'

Commit:
correct using Puppet::runtime[:http] with URI

ImpBY committed:
Merge pull request #2 from wgsateam/optimize

correct using Puppet::runtime[:http] with URI

Commit:
correct using Puppet::runtime[:http] with URI #2
